File: src/main/java/ltdjms/discord/shared/SchemaMigrationException.java
package ltdjms.discord.shared;

/**
 * Thrown when automatic schema migration detects a destructive change that cannot be applied
 * safely.
 */
public class SchemaMigrationException extends RuntimeException {

  public SchemaMigrationException(String message) {
    super(message);
  }

  public SchemaMigrationException(String message, Throwable cause) {
    super(message, cause);
  }
}

File: packages/shared/src/infra/database/schema-migration-exception.ts
/**
 * Thrown when database schema migration fails.
 * Matches Java SchemaMigrationException.
 */
export class SchemaMigrationException extends Error {
  constructor(message: string, cause?: Error) {
    super(message, { cause });
    this.name = 'SchemaMigrationException';
  }
}
